library/common/network/configurator.cc

```diff
@@ -283,15 +294,31 @@ InterfacePair Configurator::getActiveAlternateInterface(envoy_network_t network,
     // WiFi should always support multicast, and will not be point-to-point.
     auto interfaces =
         enumerateInterfaces(family, IFF_UP | IFF_MULTICAST, IFF_LOOPBACK | IFF_POINTOPOINT);
-    return interfaces.size() > 0 ? interfaces[0] : std::make_pair("", nullptr);
+    for (const auto& interface : interfaces) {
+      // Look for interface with name that matches the expected prefix.
+      // TODO(goaway): This is quite brittle. It would be an improvement to:
+      //   1) Improve the scoping via flags.
+      //   2) Prioritize interfaces by prefix instead of simply filtering them.
+      if (std::get<const std::string>(interface).rfind(WlanPrefix, 0) == 0) {
```

Comment:
I'd have used `absl::StartsWith`, but this works

Reply:
`absl::StartsWith` would have been a lot easier to understand this code as an inexperienced C++ programmer.

Reply:
Updated to use absl::StartsWith for clarity.